core/include/qrsignatures.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+/**
+ * signature verification wrapper for trezor crypto util.
+ * Input should be raw bytes(unhashed). This function will hash input with sha256
+ * before verification.
+ * @return true on successful signature verification and false otherwise.
+ */
+bool check_signature(unsigned char * data, uint32_t data_len, unsigned char * signature);
```

Comment:
nit: use `uint8_t` instead of `unsigned char` to homogenize with the rest of function prototypes elsewhere (similarly prefer `size_t` to `data_len`)

nit: use const qualifier if suitable.

nit: any reason not to use `Result` as a return type, as used elsewhere?

nit: should we choose a more specific name like `check_qrsignature` just to distinguish it from other signatures elsewhere in the code?

Reply:
Thanks for the suggestions. I think I am taking all but this:

I think the error that we want to propogate to the user is `sig not verified`. 

So maybe adding return code is not required. That's why I chose bool.